### ISSUE TITLE
Temporary failure resolving 'archive.ubuntu.com' when installing build-essential

### ISSUE THREAD

**Issue by Woodykaixa** [microsoft/WSL#7081]
### Windows Build Number

Microsoft Windows [Version 10.0.19042.1052]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

4.19.128-microsoft-standard

### Distro Version

Ubuntu 20.04

### Repro Steps

I have just installed WSL2

I have just do one thing, running `apt install build-essential`

### Expected Behavior

Install build-essential

### Actual Behavior

apt have detected packages, but cannot install

**Comment by Woodykaixa** [comment]
btw, when installing WSL2, it prompts ''The Windows Subsystem for Linux instance has terminated.", and I cannot set a username. But it finally succeed after a few trails

**Comment by therealkenc** [comment]
/dupe #6404 #5971, others. Canonical solution is putting `nameserver 8.8.8.8` in `/etc/resolve.conf` which helps in some scenarios where the DNS lookup fails.

**Comment by codemasterv** [comment]
Just going to point out if windows requires a restart for updates WLS will throw that "Temporary failure resolving archive.ubuntu.com" error.

Reboot and update windows and it will correct wls update issues if you have already done the 8.8.8.8 fix and its still erroring

**Comment by universalamateur** [comment]
This article was helpful for me to resolve my issues: https://wheatevo.com/wsl-2-and-vpn/

**Comment by Santosjordi** [comment]
I had this problem and figured out that when connected to a corporate VPN, WSL2 fails to resolve hostnames because it cannot communicate with public DNS servers (like 8.8.8.8) which are blocked by the VPN tunnel, and it fails to correctly use the corporate DNS servers.

My fix was to manually configure WSL2 to use the corporate DNS server IPs by copying the ones Windows host is currently using.

```powershell
ipconfig /all
```
And add them to:
```bash
sudo nano /etc/resolv.conf
```

As

```conf
nameserver <Corporate DNS IP 1>
nameserver <Corporate DNS IP 2>
```
Might need to restart WSL for it to work.
